File: TeamCode/src/main/java/org/hermitsocialclub/pandemicpanic/telecat/FileLogger.java
package org.hermitsocialclub.pandemicpanic.telecat;

import android.os.Environment;

import java.io.File;
import java.io.FileWriter;
import java.io.IOException;

public class FileLogger {

    public static void clearFile(String filepath) {
        try {
            File file = new File(Environment.getExternalStorageDirectory().getPath() + File.separator + "DataDumpLogs" + File.separator + filepath);
            file.getParentFile().mkdirs();
            if (!file.exists()) file.createNewFile();
            FileWriter writer = new FileWriter(file);
            writer.write("");
            writer.flush();
            writer.close();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public static void logToFile(String filepath, String nextLineToLog) {
        try {
            File file = new File(Environment.getExternalStorageDirectory().getPath() + File.separator + "DataDumpLogs" + File.separator + filepath);
            file.getParentFile().mkdirs();
            if (!file.exists()) file.createNewFile();
            FileWriter writer = new FileWriter(file, true);
            writer.append(nextLineToLog);
            writer.flush();
            writer.close();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

}

File: TeamCode/src/main/kotlin/org/hermitsocialclub/telecat/FileLogger.kt
package org.hermitsocialclub.telecat

import android.os.Environment
import java.io.File
import java.io.FileWriter
import java.io.IOException

object FileLogger {

    fun clearFile(filepath: String) {
        try {
            val file =
                File(Environment.getExternalStorageDirectory().path + File.separator + "DataDumpLogs" + File.separator + filepath)
            file.parentFile.mkdirs()
            if (!file.exists()) file.createNewFile()
            val writer = FileWriter(file)
            writer.write("")
            writer.flush()
            writer.close()
        } catch (e: IOException) {
            e.printStackTrace()
        }
    }

    fun logToFile(filepath: String, nextLineToLog: String?) {
        try {
            val file =
                File(Environment.getExternalStorageDirectory().path + File.separator + "DataDumpLogs" + File.separator + filepath)
            file.parentFile.mkdirs()
            if (!file.exists()) file.createNewFile()
            val writer = FileWriter(file, true)
            writer.append(nextLineToLog)
            writer.flush()
            writer.close()
        } catch (e: IOException) {
            e.printStackTrace()
        }
    }
}